### PR TITLE
UI: pki import issuer

### PR DESCRIPTION
**Pull request by hellobontempo**
This PR imports the issuer along with adds the serial number data to the certificate parsing helper

import-issuer

### CODE REVIEW

File: ui/app/adapters/pki/issuer.js [hunk 1]
```diff
@@ -21,6 +21,22 @@ export default class PkiIssuerAdapter extends ApplicationAdapter {
...
 
+  createRecord(store, type, snapshot) {
+    const { record, adapterOptions } = snapshot;
+    let url = this.urlForQuery(record.backend);
+    if (adapterOptions.import) {
+      url = `${url}/import/bundle`;
+    } else {
+      // TODO WIP generate
+      const type = 'root' || 'generate';
+      // record.type is internal or exported
```
Comment (hellobontempo) [inline ui/app/adapters/pki/issuer.js:32]: This is the pattern I used for generating vs importing a pki key [here](https://github.com/hashicorp/vault/blob/main/ui/app/adapters/pki/key.js#LL14)

File: ui/app/adapters/pki/issuer.js [hunk 2]
```diff
@@ -21,6 +21,22 @@ export default class PkiIssuerAdapter extends ApplicationAdapter {
     }
   }
 
+  createRecord(store, type, snapshot) {
+    const { record, adapterOptions } = snapshot;
+    let url = this.urlForQuery(record.backend);
+    if (adapterOptions.import) {
+      url = `${url}/import/bundle`;
```
Comment (hellobontempo) [inline ui/app/adapters/pki/issuer.js:28]: `issuers/import/bundle` accepts both certificate and keys, `issuers/import/cert` only accepts a certificate and errors if sent a key. After chatting with crypto, confirmed that UI should just use `/bundle`

File: ui/app/adapters/pki/issuer.js [hunk 3]
```diff
@@ -21,6 +21,22 @@ export default class PkiIssuerAdapter extends ApplicationAdapter {
     }
   }
 
+  createRecord(store, type, snapshot) {
```
Comment (hellobontempo) [inline ui/app/adapters/pki/issuer.js:24]: [Docs](https://github.com/hashicorp/vault/blob/main/website/content/api-docs/secret/pki.mdx#import-ca-certificates-and-keys)

File: ui/app/adapters/pki/issuer.js [hunk 4]
```diff
@@ -21,6 +21,22 @@ export default class PkiIssuerAdapter extends ApplicationAdapter {
...
+    const { record, adapterOptions } = snapshot;
+    let url = this.urlForQuery(record.backend);
+    if (adapterOptions.import) {
+      url = `${url}/import/bundle`;
+    } else {
+      // TODO WIP generate
+      const type = 'root' || 'generate';
```
Comment (hashishaw) [inline ui/app/adapters/pki/issuer.js:31]: The `type` here will never be `generate`, am I reading that right? If this is still WIP maybe it's best to throw an error or assertion here so we don't forget about it

File: ui/app/helpers/parse-pki-cert.js [hunk 5]
```diff
@@ -43,9 +44,13 @@ export function parseCertificate(certificateContent) {
   // field themselves are Time values.
   const expiryDate = cert?.notAfter?.value;
   const issueDate = cert?.notBefore?.value;
+  const serialNumber = Convert.ToHex(cert.serialNumber.valueBlock.valueHex)
```
Comment (hashishaw) [inline ui/app/helpers/parse-pki-cert.js:47]: 😵

**Review by hashishaw — APPROVED** [review]
Great work, thanks!!

File: ui/app/adapters/pki/issuer.js (hunk 4 above)
Comment (hellobontempo) [reply ui/app/adapters/pki/issuer.js:31]: I think i mis-typed this and it should be root or intermediate

File: ui/app/helpers/parse-pki-cert.js (hunk 5 above)
Comment (hellobontempo) [reply ui/app/helpers/parse-pki-cert.js:47]: Yeah 🥴  this took way longer than I want to admit

**Pull request merged**